metadata-protobuf/proto/App.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto2";
+
+import "proto/Video.proto";
+import "proto/Channel.proto";
+
+message AppActionMetadata {
+  optional string nonce = 1;
+  // YouTube video ID
+  optional string video_id  = 2;
+}
+
+message AppAction {
+  // ID of application
+  optional string app_id = 1;
+
+  // Metadata
+  optional AppActionMetadata metadata = 2;
+
+  // Raw metadata of wrapped action
+  oneof raw_action {
+    ContentMetadata content_metadata  = 3;
+    ChannelMetadata channel_metadata = 4;
+  }
```

Comment:
Similarly to the previous comment, https://github.com/Joystream/joystream/issues/4307 states:
> The reason raw_action is not locked to a specific set of messages (...)

In you implementation it is actually being locked to a specific set of messages: `ContentMetadata` and `ChannelMetadata`.

As I understand this should be just a raw string, the QN can try to decode it as `ContentMetadata` or `ChannelMetadata` depending on which event it processes, but a failure to decode `raw_action` shouldn't cause a failure to decode the entire `AppAction`. In case `raw_action` is not in a recognized format, it should be just treated as empty metadata, but the channel/video should still be connected to a given app.